common/src/events/marshal/marshal.types.ts

```diff
@@ -0,0 +1,13 @@
+import {z} from "zod";
+
+import type {EventPayload} from "../events.types";
+import {EventTopic} from "../events.types";
+
+export type EventFunction<Event extends EventTopic> = (data: EventPayload<Event>) => Promise<void>;
```

Comment:
Not all event functions will be `async`

```suggestion
export type EventFunction<Event extends EventTopic> = (data: EventPayload<Event>) => void | Promise<void>;
```